onnx/numpy_helper.py

```diff
@@ -118,6 +118,139 @@ def from_array(arr, name=None):  # type: (np.ndarray[Any], Optional[Text]) -> Te
     return tensor
 
 
+def to_list(sequence):  # type: (SequenceProto) -> List[Any]
+    """Converts a sequence def to a Python list.
+
+    Inputs:
+        sequence: a SequenceProto object.
+    Returns:
+        lst: the converted list.
+    """
+    lst = []
+    elem_type = sequence.elem_type
+    value_field = mapping.STORAGE_ELEMENT_TYPE_TO_FIELD[elem_type]
+    values = getattr(sequence, value_field)
+    for value in values:
+        if elem_type == Sequence.TENSOR or elem_type == Sequence.SPARSE_TENSOR:
+            lst.append(to_array(value))
+        elif elem_type == Sequence.SEQUENCE:
+            lst.append(to_list(value))
+        elif elem_type == Sequence.MAP:
+            lst.append(to_dict(value))
+        else:
+            raise TypeError("The element type in the input sequence is not supported.")
+    return lst
+
+
+def from_list(lst, name=None, dtype=np.ndarray):  # type: (List[Any], Optional[Text]) -> SequenceProto
+    """Converts a list into a sequence def.
+
+    Inputs:
+        lst: a Python list
+        name: (optional) the name of the sequence.
+        dtype: (optional) type of element in the input list, used for specifying
+                          sequence values when converting an empty list.
+    Returns:
+        sequence: the converted sequence def.
+    """
+    sequence = SequenceProto()
+    if name:
+        sequence.name = name
+
+    if len(lst) == 0:
+        elem_type = dtype
+    else:
+        elem_type = type(lst[0])
```

Comment:
this does not seem right. The elem_type here will be the type of the input 1 in the list... it is not compatible with sequence.elem_type .... this code will not work...
The type should be one of the following:
enum DataType {
    UNDEFINED = 0;
    TENSOR = 1;
    SPARSE_TENSOR = 2;
    SEQUENCE = 3;
    MAP = 4;
  }